geowebcache/core/src/test/java/org/geowebcache/layer/wms/WMSLayerTest.java

```diff
@@ -359,7 +359,7 @@ public void testHttpClientNeedsToBeCreated() throws Exception {
     @Test
     public void testCascadeGetLegendGraphics() throws Exception {
         // setup the layer
-        WMSLayer layer = createWMSLayer("image/png");
+        WMSLayer layer = createWMSLayer("image/png;charset=UTF-8");
```

Comment:
Yes.  Changed back.  However a few lines down I had to leave the "expect" with the charset because of this: https://github.com/GeoWebCache/geowebcache/blob/b4dff9c2244fe4569319fe82b453f4f82387fd5f/geowebcache/core/src/main/java/org/geowebcache/layer/wms/WMSLayer.java#L818  Should I look at removing that because we would expect WMS to only return binary results (assuming here that we are only dealing with getLegendGraphic and getMap)

Reply:
WMS can return JSON too (JSON vector tiles, SVG tiles), so some formats might use the UTF8 extra bit... but is it necessary? Probably not. Removing it wholesale it's likely to be easier.
As an alternative, you can use the mime-type classes in GWC to figure out if a format is binary or not, I presume (maybe extend MimeType with a flat reporting if a format is to be considered binary or not?)

Reply:
Added the second option